### PR TITLE
Enhance Google Play Map fragment

### LINKED ISSUE #1419: Better Map Activity
**Actual Behaviour**

Currently, only a single marker is shown on the map.

**Expected Behaviour**

The map activity could be greatly enhanced like the google I/O app.
- Marker clustering
- Better marker drawables
- Location searching
- Navigation to the location marker

**Screenshots of the issue**

mockup - 
mapactivity1

**Would you like to work on the issue?**

Yes, I would like to work on it.

### PR DESCRIPTION
**Pull request by rohanagarwal94**
Fixes #1419 

Changes: 
- Autocompletetextview for searching 
- Different colour for focused marker.
- Directions enabled through Google maps.

Screenshots for the change: 
screenshot_2017-06-01-00-21-16-721_org fossasia openevent

### CODE REVIEW

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 1]
```diff
@@ -30,13 +36,13 @@
...
-
 import io.reactivex.disposables.CompositeDisposable;
 
-public class MapsFragment extends SupportMapFragment
-        implements LocationListener, OnMapReadyCallback {
+public class MapsFragment extends Fragment implements LocationListener, OnMapReadyCallback {
     private GoogleMap mMap;
-    private LatLng location;
+    private Marker locationMarker;
+    private AutoCompleteTextView actv;
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: Better variable names please

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 2]
```diff
@@ -30,13 +36,13 @@
...
+    private List<String> actvItems = new ArrayList<>();
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: Same as above

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 3]
```diff
@@ -48,9 +54,12 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
-        super.onViewCreated(view, savedInstanceState);
-        getMapAsync(this);
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_map, container, false);
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: `view` or `root` will be better

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 4]
```diff
@@ -48,9 +54,12 @@ public void onCreate(Bundle savedInstanceState) {
...
+        actv = (AutoCompleteTextView) v.findViewById(R.id.autoCompleteTextView);
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: Use Butterknife

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 5]
```diff
@@ -72,43 +83,59 @@ public void onMapReady(GoogleMap map) {
...
             location = new LatLng(latitude, longitude);
+            locationName = microlocation.getName();
+            actvItems.add(locationName);
 
-            marker = mMap.addMarker(new MarkerOptions().position(location).title(microlocation.getName()));
+            marker = handleMarkerEvents(location, locationName);
             builder.include(marker.getPosition());
         }
 
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(),
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: When adding a large block, consider transferring to a function

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 6]
```diff
@@ -124,6 +151,24 @@ private int dpToPx(int dp) {
         return (int) (dp * Resources.getSystem().getDisplayMetrics().density);
     }
 
+    private Marker handleMarkerEvents(LatLng location, String locationTitle)
+    {
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: Use Java style bracket notation

File: android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java [hunk 7]
```diff
@@ -153,9 +198,15 @@ public void onDestroyView() {
             compositeDisposable.dispose();
     }
 
+    private void launchDirections() {
+        // Build intent to start Google Maps directions
+
```
Comment (iamareebjamal) [inline android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java]: Remove extra line

File: android/app/src/googleplay/res/layout/fragment_map.xml [hunk 8]
```diff
@@ -1,5 +1,33 @@
 <?xml version="1.0" encoding="utf-8"?>
-<com.google.android.gms.maps.MapView xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/map"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (iamareebjamal) [inline android/app/src/googleplay/res/layout/fragment_map.xml]: RelativeLayout with single child does not make sense. Make FrameLayout root layout

**Comment by rohanagarwal94** [comment]
@iamareebjamal vector icons not available. Please review.

**Comment by iamareebjamal** [comment]
@rohanagarwal94 PNG is a raster format. SVG is vector -> And its counterpart in Android - Vector Drawables

Please use them instead, look at other icons in the project present in `drawable` folder

**Comment by rohanagarwal94** [comment]
@iamareebjamal these icons are taken directly from the google I/O app. They aren't available s vector assets in the studio. Any other alternative?

**Comment by iamareebjamal** [comment]
Use any location based icon from the sites I gave. There are plenty

**Comment by rohanagarwal94** [comment]
@iamareebjamal updated.

**Comment by iamareebjamal** [comment]
Views is a shared class, what you can do is just create another method in GPlay version as no other class will call vectorToBitmap with `BitmapDescriptor` as return type

**Comment by rohanagarwal94** [comment]
@iamareebjamal I haven't even touched the fdroid build. It is still the googleplay flavour.

**Comment by iamareebjamal** [comment]
Also, when you are done. Please add new screenshots. Thanks. 

Nice work :+1:

**Comment by iamareebjamal** [comment]
@rohanagarwal94 You added the BitmapDescriptor in `Views` class, which is shared between two variants, and it is a google library class not available in F-Droid build

**Comment by rohanagarwal94** [comment]
@iamareebjamal updating it :+1:

**Comment by iamareebjamal** [comment]
Also, please squash commits when build passes

**Pull request merged**